File: cp/A_2023.cpp
#include <bits/stdc++.h>
using namespace std;
#define ll long long
#define mod 1000000007

int main()
{
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);
    ll ttt = 1;
    cin >> ttt;
    while (ttt--)
    {
        int n, k;
        cin >> n >> k;

        ll mult = 1;
        for (int i = 0; i < n; i++)
        {
            ll b;
            cin >> b;
            mult *= b;
        }
        if (mult > 2023)
        {
            cout << "NO\n";
        }
        else
        {
            if (2023 % mult == 0)
            {
                cout << "YES\n";
                cout << 2023 / mult << " ";
                for (int i = 0; i < k - 1; i++)
                {
                    cout << 1 << " ";
                }
                cout << "\n";
            }
            else
            {
                cout << "NO\n";
            }
        }
    }
    return 0;
}

File: cp/A_2023.py
def main():
    n, k = map(int, input().split())
    b = list(map(int, input().split()))

    mult = b[0]
    for i in range(1, n):
        mult *= b[i]
    
    if mult > 2023:
        print("NO")
    elif 2023 % mult == 0:
        print("YES")
        print(2023 // mult, end=" ")
        for _ in range(1, k):
            print(1, end=" ");
        print()
    else:
        print("NO")


if __name__=="__main__":
    tt = int(input())
    for _ in range(tt):
        main()